weather_tools.py

```diff
@@ -0,0 +1,131 @@
+'''This code extracts weather data from the NOAA ASOS system : https://www.weather.gov/asos/asostech
+
+All the ASOS data can be accessed via ftp from their server : ftp://ftp.ncdc.noaa.gov/pub/data/noaa
+
+Within this there are folders for each years data named 'yyyy'.
+
+Inside each year there are gzipped files for each weather station in the network. 
+
+The name of each file follows a pattern given by: AAAAAA-BBBBBB-YYYY.gz. For example,
+723150-03812-2006 corresponds with USAF number 723150 and WBAN number 03812.
+
+USAF and WBAN are identifiers for each station and their information is available in the **ASOS_stations.txt** file
+    
+YYYY stands for year'''
+
+
+
+# Import statements
+
+import pandas as pd
+import numpy as np
+import re
+import matplotlib.pyplot as plt
+pd.plotting.register_matplotlib_converters()
+
+'''StationID dictionary 
+I'm defining the stationID to be the string "USAF identifier-WBAN identifier"
+This chunk of code generates a dict where keys are state names, values are stationID associated with that state.'''
+
+
+def generate_stationID_dict():
+    '''I'm defining the stationID to be the string "USAF identifier-WBAN identifier"
+      This chunk of code generates a dict where keys are state names and  
+      values are stationID associated with that state. '''
+    
+    #filepath_stations ="/Users/NikhilStuff/Desktop/social-master/ASOS_stations.txt"
+    filepath_stations = "ftp://ftp.ncdc.noaa.gov/pub/data/noaa/isd-history.txt"
+    cols_to_use = ['USAF', 'WBAN','CTRY', 'ST']
+    df_stations = pd.read_fwf(filepath_stations, skiprows = 20, header =0, usecols = cols_to_use)
+    df_stations_US = df_stations[df_stations.CTRY =='US'].copy()
+    df_stations_US.WBAN = df_stations_US.WBAN.astype('int64')
+    df_stations_US.dropna(inplace =True)
+    df_stations_US = df_stations_US[df_stations_US.WBAN !=99999].copy()
+
+    def my_func(row):
+        return str(row.USAF)+ '-' + str(row.WBAN).zfill(5)
+    df_stations_US['stationID'] = df_stations_US.apply(my_func, axis =1)
+
+    states = pd.unique(df_stations_US.ST.values)
+    stations_in_state = {}
+    for i in range(len(states)):
+        stations_in_state[states[i]] = list(df_stations_US[df_stations_US.ST == states[i]].stationID.values)
+    return stations_in_state
+stations_in_state = generate_stationID_dict()
+
+
+
+def df_from_file(stationID,YEAR):
```

Comment:
All caps variable names are usually reserved for constants. I suggest changing YEAR to year everywhere